packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -466,6 +466,15 @@ abstract class FlutterCommand extends Command<Null> {
       if (!fs.isFileSync(targetPath))
         throw new ToolExit('Target file "$targetPath" not found.');
     }
+
+    final bool dynamicFlag = argParser.options.containsKey('dynamic')
+        ? argResults['dynamic'] : false;
+    final String compilationTraceFilePath = argParser.options.containsKey('precompile')
+        ? argResults['precompile'] : null;
+    if (compilationTraceFilePath != null && getBuildMode() == BuildMode.debug)
+      throw new ToolExit('Error: --precompile is not allowed when --debug is specified.');
+    if (compilationTraceFilePath != null && !dynamicFlag)
+      throw new ToolExit('Error: --precompile is allowed only when --dynamic is specified.');
```

Comment:
Do you need a check that enforces compilationTraceFilePath flag if dynamicFlag is specified? Otherwise, what is the default value for compilationTraceFilePath?

Reply:
compilationTraceFilePath is always optional, not required when --dynamic is used.

Reply:
Where do we initialize to some default value? Which seem to be `trace.txt` in the past

Reply:
The default is null, which will cause precompilation to be disabled.

Reply:
Okay, got it. Thanks.